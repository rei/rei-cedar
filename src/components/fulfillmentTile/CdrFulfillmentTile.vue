<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import type * as CSS from 'csstype';
import mapClasses from '../../utils/mapClasses';
import CdrSurfaceSelection from '../surfaceSelection/CdrSurfaceSelection.vue';
import CdrFulfillmentTileHeader from './CdrFulfillmentTileHeader.vue';
import CdrUtilitySans, { type utilitySansTextProps } from '../text/presets/CdrUtilitySans.vue';
import { CdrFulfillmentTileProps } from '../../types/interfaces'

/** Tile component for displaying a button with optional icon at the top right */

defineOptions({ name: 'CdrFulfillmentTile' });

const props = withDefaults(defineProps<CdrFulfillmentTileProps>(), {
  typeScale: '0',
});

const style = useCssModule();
const baseClass = 'cdr-fulfillment-tile';

// Manages the props passed along to CdrGenericButton
const rootProps = computed(() => {
  const classes = [baseClass];
  const inlineStyles: CSS.Properties = {};

  return {
    ...props,
    style: inlineStyles,
    class: mapClasses(style, ...classes),
  };
});

const headerProps = computed(() => ({
  typeScale: props.typeScale,
  class: style[`${baseClass}__header`],
}));

const bodyProps = computed<utilitySansTextProps>(() => ({
  scale: '-1',
  tag: 'div',
  class: {
    [style[`${baseClass}__content`]]: true,
    [style[`${baseClass}__content--body`]]: true,
  },
}));

const footerProps = computed<utilitySansTextProps>(() => ({
  scale: '-1',
  tag: 'div',
  class: {
    [style[`${baseClass}__content`]]: true,
    [style[`${baseClass}__content--footer`]]: true,
  },
}));
</script>

<template>
  <CdrSurfaceSelection v-bind="rootProps">
    <CdrFulfillmentTileHeader
      v-if="$slots.label"
      v-bind="headerProps"
    >
      <template
        v-if="$slots['icon-left']"
        #icon-left
      >
        <slot name="icon-left" />
      </template>
      <template #label>
        <slot name="label" />
      </template>
      <template
        v-if="$slots['icon-right']"
        #icon-right
      >
        <slot name="icon-right" />
      </template>
    </CdrFulfillmentTileHeader>
    <CdrUtilitySans
      v-if="$slots.body"
      v-bind="bodyProps"
    >
      <slot name="body" />
    </CdrUtilitySans>
    <CdrUtilitySans
      v-if="$slots.footer"
      v-bind="footerProps"
    >
      <slot name="footer" />
    </CdrUtilitySans>
  </CdrSurfaceSelection>
</template>

<style lang="scss" module src="./styles/CdrFulfillmentTile.module.scss"></style>
