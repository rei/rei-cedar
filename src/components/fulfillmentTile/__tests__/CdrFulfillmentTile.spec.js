import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrFulfillmentTile from '../CdrFulfillmentTile.vue';

describe('CdrFulfillmentTile', () => {
  describe('snapshot test', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrFulfillmentTile, { props: { checked: false } });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('component unit tests', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrFulfillmentTile, { props: { checked: false } });
    });

    describe('default component with no configuration', () => {
      it('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });
      it('sets the default tag prop correctly', () => {
        expect(wrapper.element.tagName).toBe('BUTTON');
      });
      it('base classes only', () => {
        expect(wrapper.classes()).toEqual(['cdr-fulfillment-tile']);
      });
    });
  });

  it('supports icon left configuration', () => {
    const wrapper = mount(CdrFulfillmentTile, {
      props: { checked: false },
      slots: {
        'icon-left': 'X',
        label: 'Label slot',
        body: 'Body slot',
        footer: 'Footer slot',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('supports  icon right configuration', () => {
    const wrapper = mount(CdrFulfillmentTile, {
      props: { checked: false },
      slots: {
        'icon-right': 'X',
        label: 'Label slot',
        body: 'Body slot',
        footer: 'Footer slot',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('shows header', () => {
    const wrapper = mount(CdrFulfillmentTile, {
      props: { checked: false },
      slots: {
        label: 'Label slot',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.cdr-fulfillment-tile-header').text()).toBe('Label slot');
  });

  it('shows body', () => {
    const wrapper = mount(CdrFulfillmentTile, {
      props: { checked: false },
      slots: {
        label: 'Label slot',
        body: 'Body slot',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.cdr-fulfillment-tile-content--stretch').text()).toBe('Body slot');

  });

  it('shows footer', () => {
    const wrapper = mount(CdrFulfillmentTile, {
      props: { checked: false },
      slots: {
        label: 'Label slot',
        footer: 'Footer slot',
      },
    });
    expect(wrapper.find('.cdr-fulfillment-tile-content').text()).toBe('Footer slot');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('is loading', () => {
    const wrapper = mount(CdrFulfillmentTile, {
      props: { checked: false, loading: true },
      slots: {
        label: 'Label slot',
        footer: 'Footer slot',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.cdr-fulfillment-tile__loading').exists()).toBe(true);
    expect(wrapper.find('.cdr-fulfillment-tile-header').exists()).toBe(false);
    expect(wrapper.find('.cdr-fulfillment-tile-content').exists()).toBe(false);
  });
});
