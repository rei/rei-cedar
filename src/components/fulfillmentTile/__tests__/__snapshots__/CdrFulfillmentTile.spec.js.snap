// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrFulfillmentTile > component unit tests > default component with no configuration > renders correctly 1`] = `
<button
  aria-checked="false"
  class="cdr-fulfillment-tile cdr-fulfillment-tile--modifier-default"
  modifier="default"
  orientation="vertical"
  role="radio"
  tag="button"
  type="button"
>
  <div
    class="cdr-fulfillment-tile__inner"
  >
    <div
      class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
    >
      <!-- @slot Place elements in the default slot. -->
      
      <!--v-if-->
      <span
        class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
        strong="true"
        style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
      >
        
        <!-- @slot Header content that is still visible during loading. -->
        
        <!-- @slot Header content that is still visible during loading. -->
        
        Header
        
        
        
      </span>
      <!--v-if-->
      
    </div>
    <div
      class="cdr-fulfillment-tile__main"
    >
      <div
        class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--vertical cdr-fulfillment-tile__layout"
      >
        <!-- @slot Place elements in the default slot. -->
        
        
      </div>
      <div
        class="cdr-fulfillment-tile__loading"
      >
        <!-- @slot Custom content when component is loading. -->
        
        <div
          aria-busy="true"
          aria-live="polite"
          class="cdr-skeleton"
        >
          <!-- @slot CdrSkeleton content (CdrSkeletonBone components) -->
          
          <div
            class="cdr-skeleton-bone cdr-skeleton-bone--line cdr-skeleton-bone__shimmer"
          />
          
        </div>
        
      </div>
    </div>
  </div>
</button>
`;

exports[`CdrFulfillmentTile > is loading 1`] = `
<button
  aria-checked="false"
  class="cdr-fulfillment-tile cdr-fulfillment-tile--modifier-default"
  data-loading="true"
  modifier="default"
  orientation="vertical"
  role="radio"
  tag="button"
  type="button"
>
  <div
    class="cdr-fulfillment-tile__inner"
  >
    <div
      class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
    >
      <!-- @slot Place elements in the default slot. -->
      
      <!--v-if-->
      <span
        class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
        strong="true"
        style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
      >
        
        <!-- @slot Header content that is still visible during loading. -->
        
        <!-- @slot Header content that is still visible during loading. -->
        
        Label slot
        
        
        
      </span>
      <!--v-if-->
      
    </div>
    <div
      class="cdr-fulfillment-tile__main"
    >
      <div
        class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--vertical cdr-fulfillment-tile__layout"
      >
        <!-- @slot Place elements in the default slot. -->
        
        <!--v-if-->
        <div
          class="cdr-body cdr-fulfillment-tile-content"
          style="--cdr-body-font-size: var(--cdr-type-scale--1); --cdr-body-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-body-line-height-ratio)); --cdr-body-font-weight: 400;"
        >
          
          <!-- @slot Where all default content should be placed. -->
          
          <!-- @slot Footer content will be at the bottom of the component. -->
          
          Footer slot
          
          
          
        </div>
        
      </div>
      <div
        class="cdr-fulfillment-tile__loading"
      >
        <!-- @slot Custom content when component is loading. -->
        
        <div
          aria-busy="true"
          aria-live="polite"
          class="cdr-skeleton"
        >
          <!-- @slot CdrSkeleton content (CdrSkeletonBone components) -->
          
          <div
            class="cdr-skeleton-bone cdr-skeleton-bone--line cdr-skeleton-bone__shimmer"
          />
          
        </div>
        
      </div>
    </div>
  </div>
</button>
`;

exports[`CdrFulfillmentTile > shows body 1`] = `
<button
  aria-checked="false"
  class="cdr-fulfillment-tile cdr-fulfillment-tile--modifier-default"
  modifier="default"
  orientation="vertical"
  role="radio"
  tag="button"
  type="button"
>
  <div
    class="cdr-fulfillment-tile__inner"
  >
    <div
      class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
    >
      <!-- @slot Place elements in the default slot. -->
      
      <!--v-if-->
      <span
        class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
        strong="true"
        style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
      >
        
        <!-- @slot Header content that is still visible during loading. -->
        
        <!-- @slot Header content that is still visible during loading. -->
        
        Label slot
        
        
        
      </span>
      <!--v-if-->
      
    </div>
    <div
      class="cdr-fulfillment-tile__main"
    >
      <div
        class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--vertical cdr-fulfillment-tile__layout"
      >
        <!-- @slot Place elements in the default slot. -->
        
        <div
          class="cdr-body cdr-fulfillment-tile-content cdr-fulfillment-tile-content--stretch"
          style="--cdr-body-font-size: var(--cdr-type-scale--2); --cdr-body-line-height: calc(var(--cdr-type-scale--2) * var(--cdr-body-line-height-ratio)); --cdr-body-font-weight: 400;"
        >
          
          <!-- @slot Where all default content should be placed. -->
          
          <!-- @slot Default font size is a step down. Placed just below the header. -->
          
          Body slot
          
          
          
        </div>
        <!--v-if-->
        
      </div>
      <div
        class="cdr-fulfillment-tile__loading"
      >
        <!-- @slot Custom content when component is loading. -->
        
        <div
          aria-busy="true"
          aria-live="polite"
          class="cdr-skeleton"
        >
          <!-- @slot CdrSkeleton content (CdrSkeletonBone components) -->
          
          <div
            class="cdr-skeleton-bone cdr-skeleton-bone--line cdr-skeleton-bone__shimmer"
          />
          
        </div>
        
      </div>
    </div>
  </div>
</button>
`;

exports[`CdrFulfillmentTile > shows footer 1`] = `
<button
  aria-checked="false"
  class="cdr-fulfillment-tile cdr-fulfillment-tile--modifier-default"
  modifier="default"
  orientation="vertical"
  role="radio"
  tag="button"
  type="button"
>
  <div
    class="cdr-fulfillment-tile__inner"
  >
    <div
      class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
    >
      <!-- @slot Place elements in the default slot. -->
      
      <!--v-if-->
      <span
        class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
        strong="true"
        style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
      >
        
        <!-- @slot Header content that is still visible during loading. -->
        
        <!-- @slot Header content that is still visible during loading. -->
        
        Label slot
        
        
        
      </span>
      <!--v-if-->
      
    </div>
    <div
      class="cdr-fulfillment-tile__main"
    >
      <div
        class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--vertical cdr-fulfillment-tile__layout"
      >
        <!-- @slot Place elements in the default slot. -->
        
        <!--v-if-->
        <div
          class="cdr-body cdr-fulfillment-tile-content"
          style="--cdr-body-font-size: var(--cdr-type-scale--1); --cdr-body-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-body-line-height-ratio)); --cdr-body-font-weight: 400;"
        >
          
          <!-- @slot Where all default content should be placed. -->
          
          <!-- @slot Footer content will be at the bottom of the component. -->
          
          Footer slot
          
          
          
        </div>
        
      </div>
      <div
        class="cdr-fulfillment-tile__loading"
      >
        <!-- @slot Custom content when component is loading. -->
        
        <div
          aria-busy="true"
          aria-live="polite"
          class="cdr-skeleton"
        >
          <!-- @slot CdrSkeleton content (CdrSkeletonBone components) -->
          
          <div
            class="cdr-skeleton-bone cdr-skeleton-bone--line cdr-skeleton-bone__shimmer"
          />
          
        </div>
        
      </div>
    </div>
  </div>
</button>
`;

exports[`CdrFulfillmentTile > shows header 1`] = `
<button
  aria-checked="false"
  class="cdr-fulfillment-tile cdr-fulfillment-tile--modifier-default"
  modifier="default"
  orientation="vertical"
  role="radio"
  tag="button"
  type="button"
>
  <div
    class="cdr-fulfillment-tile__inner"
  >
    <div
      class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
    >
      <!-- @slot Place elements in the default slot. -->
      
      <!--v-if-->
      <span
        class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
        strong="true"
        style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
      >
        
        <!-- @slot Header content that is still visible during loading. -->
        
        <!-- @slot Header content that is still visible during loading. -->
        
        Label slot
        
        
        
      </span>
      <!--v-if-->
      
    </div>
    <div
      class="cdr-fulfillment-tile__main"
    >
      <div
        class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--vertical cdr-fulfillment-tile__layout"
      >
        <!-- @slot Place elements in the default slot. -->
        
        
      </div>
      <div
        class="cdr-fulfillment-tile__loading"
      >
        <!-- @slot Custom content when component is loading. -->
        
        <div
          aria-busy="true"
          aria-live="polite"
          class="cdr-skeleton"
        >
          <!-- @slot CdrSkeleton content (CdrSkeletonBone components) -->
          
          <div
            class="cdr-skeleton-bone cdr-skeleton-bone--line cdr-skeleton-bone__shimmer"
          />
          
        </div>
        
      </div>
    </div>
  </div>
</button>
`;

exports[`CdrFulfillmentTile > snapshot test > renders correctly 1`] = `
<button
  aria-checked="false"
  class="cdr-fulfillment-tile cdr-fulfillment-tile--modifier-default"
  modifier="default"
  orientation="vertical"
  role="radio"
  tag="button"
  type="button"
>
  <div
    class="cdr-fulfillment-tile__inner"
  >
    <div
      class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
    >
      <!-- @slot Place elements in the default slot. -->
      
      <!--v-if-->
      <span
        class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
        strong="true"
        style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
      >
        
        <!-- @slot Header content that is still visible during loading. -->
        
        <!-- @slot Header content that is still visible during loading. -->
        
        Header
        
        
        
      </span>
      <!--v-if-->
      
    </div>
    <div
      class="cdr-fulfillment-tile__main"
    >
      <div
        class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--vertical cdr-fulfillment-tile__layout"
      >
        <!-- @slot Place elements in the default slot. -->
        
        
      </div>
      <div
        class="cdr-fulfillment-tile__loading"
      >
        <!-- @slot Custom content when component is loading. -->
        
        <div
          aria-busy="true"
          aria-live="polite"
          class="cdr-skeleton"
        >
          <!-- @slot CdrSkeleton content (CdrSkeletonBone components) -->
          
          <div
            class="cdr-skeleton-bone cdr-skeleton-bone--line cdr-skeleton-bone__shimmer"
          />
          
        </div>
        
      </div>
    </div>
  </div>
</button>
`;

exports[`CdrFulfillmentTile > supports  icon right configuration 1`] = `
<button
  aria-checked="false"
  class="cdr-fulfillment-tile cdr-fulfillment-tile--modifier-default"
  modifier="default"
  orientation="vertical"
  role="radio"
  tag="button"
  type="button"
>
  <div
    class="cdr-fulfillment-tile__inner"
  >
    <div
      class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
    >
      <!-- @slot Place elements in the default slot. -->
      
      <!--v-if-->
      <span
        class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
        strong="true"
        style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
      >
        
        <!-- @slot Header content that is still visible during loading. -->
        
        <!-- @slot Header content that is still visible during loading. -->
        
        Label slot
        
        
        
      </span>
      <span
        class="cdr-fulfillment-tile-header__icon"
      >
        <!-- @slot Place an icon to the right of the header content -->
        
        <!-- @slot Place an icon to the right of the header content -->
        
        X
        
        
      </span>
      
    </div>
    <div
      class="cdr-fulfillment-tile__main"
    >
      <div
        class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--vertical cdr-fulfillment-tile__layout"
      >
        <!-- @slot Place elements in the default slot. -->
        
        <div
          class="cdr-body cdr-fulfillment-tile-content cdr-fulfillment-tile-content--stretch"
          style="--cdr-body-font-size: var(--cdr-type-scale--2); --cdr-body-line-height: calc(var(--cdr-type-scale--2) * var(--cdr-body-line-height-ratio)); --cdr-body-font-weight: 400;"
        >
          
          <!-- @slot Where all default content should be placed. -->
          
          <!-- @slot Default font size is a step down. Placed just below the header. -->
          
          Body slot
          
          
          
        </div>
        <div
          class="cdr-body cdr-fulfillment-tile-content"
          style="--cdr-body-font-size: var(--cdr-type-scale--1); --cdr-body-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-body-line-height-ratio)); --cdr-body-font-weight: 400;"
        >
          
          <!-- @slot Where all default content should be placed. -->
          
          <!-- @slot Footer content will be at the bottom of the component. -->
          
          Footer slot
          
          
          
        </div>
        
      </div>
      <div
        class="cdr-fulfillment-tile__loading"
      >
        <!-- @slot Custom content when component is loading. -->
        
        <div
          aria-busy="true"
          aria-live="polite"
          class="cdr-skeleton"
        >
          <!-- @slot CdrSkeleton content (CdrSkeletonBone components) -->
          
          <div
            class="cdr-skeleton-bone cdr-skeleton-bone--line cdr-skeleton-bone__shimmer"
          />
          
        </div>
        
      </div>
    </div>
  </div>
</button>
`;

exports[`CdrFulfillmentTile > supports icon left configuration 1`] = `
<button
  aria-checked="false"
  class="cdr-fulfillment-tile cdr-fulfillment-tile--modifier-default"
  modifier="default"
  orientation="vertical"
  role="radio"
  tag="button"
  type="button"
>
  <div
    class="cdr-fulfillment-tile__inner"
  >
    <div
      class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
    >
      <!-- @slot Place elements in the default slot. -->
      
      <span
        class="cdr-fulfillment-tile-header__icon"
      >
        <!-- @slot Place an icon to the left of the header content -->
        
        <!-- @slot Place an icon to the left of the header content -->
        
        X
        
        
      </span>
      <span
        class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
        strong="true"
        style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
      >
        
        <!-- @slot Header content that is still visible during loading. -->
        
        <!-- @slot Header content that is still visible during loading. -->
        
        Label slot
        
        
        
      </span>
      <!--v-if-->
      
    </div>
    <div
      class="cdr-fulfillment-tile__main"
    >
      <div
        class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--vertical cdr-fulfillment-tile__layout"
      >
        <!-- @slot Place elements in the default slot. -->
        
        <div
          class="cdr-body cdr-fulfillment-tile-content cdr-fulfillment-tile-content--stretch"
          style="--cdr-body-font-size: var(--cdr-type-scale--2); --cdr-body-line-height: calc(var(--cdr-type-scale--2) * var(--cdr-body-line-height-ratio)); --cdr-body-font-weight: 400;"
        >
          
          <!-- @slot Where all default content should be placed. -->
          
          <!-- @slot Default font size is a step down. Placed just below the header. -->
          
          Body slot
          
          
          
        </div>
        <div
          class="cdr-body cdr-fulfillment-tile-content"
          style="--cdr-body-font-size: var(--cdr-type-scale--1); --cdr-body-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-body-line-height-ratio)); --cdr-body-font-weight: 400;"
        >
          
          <!-- @slot Where all default content should be placed. -->
          
          <!-- @slot Footer content will be at the bottom of the component. -->
          
          Footer slot
          
          
          
        </div>
        
      </div>
      <div
        class="cdr-fulfillment-tile__loading"
      >
        <!-- @slot Custom content when component is loading. -->
        
        <div
          aria-busy="true"
          aria-live="polite"
          class="cdr-skeleton"
        >
          <!-- @slot CdrSkeleton content (CdrSkeletonBone components) -->
          
          <div
            class="cdr-skeleton-bone cdr-skeleton-bone--line cdr-skeleton-bone__shimmer"
          />
          
        </div>
        
      </div>
    </div>
  </div>
</button>
`;
